### PR TITLE
Bio 1876 adapt to nf core sarek rb

### PR DESCRIPTION
**Pull request by b97pla**
This PR introduces breaking changes w.r.t. handling the Sarek pipeline. 

The Sarek engine has been rewritten to handle Sarek >= 2.6. Here are the main changes:

- Only the main Sarek workflow is invoked
- Separated config sections for nextflow and sarek 
- Adjust the genome_base_path parameter for GRCh37 and GRCh38
- By default, the resume flag is used
- Added a script to parse results and update Charon independently from the local tracking database
- Added some scripts to build docker and singularity containers (won't be able to run SLURM from the container though)

### CODE REVIEW

**Comment by matrulda** [comment]
Wow, great work! Just wanted to drop a comment that this review is on my to-do list, other things has gotten in the way. It's quite a big PR, so I'll have to spend some time on it.

File: ngi_pipeline/tests/engines/sarek/models/test_sarek.py [hunk 1]
```diff
@@ -102,29 +104,86 @@ def _alignment_status_wrapper(status, *args):
...
-        for key in config.keys():
-            self.assertEqual(config[key], sarek_analysis.sarek_config[key])
-        self.assertEqual(SarekAnalysis.DEFAULT_CONFIG["nf_path"], sarek_analysis.sarek_config["nf_path"])
-        self.assertEqual(SarekAnalysis.DEFAULT_CONFIG["sarek_path"], sarek_analysis.sarek_config["sarek_path"])
+        for key in config["sarek"].keys():
+            self.assertEqual(config["sarek"][key], sarek_analysis.sarek_config[key])
+        for key in config["nextflow"].keys():
+            self.assertEqual(config["nextflow"][key], sarek_analysis.nextflow_config[key])
+
+        del(config["nextflow"]["profile"])
```
Comment (matrulda) [inline ngi_pipeline/tests/engines/sarek/models/test_sarek.py:127]: This test is really long and a bit hard to follow, I think it could benefit from being divided into different tests. For example this part when testing if the default value is used.

File: requirements-dev.txt [hunk 2]
```diff
@@ -1,7 +1,7 @@
 -r requirements.txt
-codecov==2.0.9
+codecov
```
Comment (matrulda) [inline requirements-dev.txt:2]: Why do you remove the specific versions? I had problems with running the tests since I had a too old version of PyAML.

File: docker/ngi_config.yaml [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    #jobNative:
+    #    - arg1
+    #    - arg2
+    #    - arg3
+    # TODO: Piper module adds setupfilecreator to the path; unsure if this works or if complete path is required.  
+    path_to_setupfilecreator: setupfilecreator
+    gatk_key: "/path/to/gatk.key"
+
+sarek:
+    tag: 2.6
```
Comment (aanil) [inline docker/ngi_config.yaml:33]: Sarek 2.6.1 has now been released with GATK updated. Version 2.6 didn't work completely when we tried to run it on Irma.

File: docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,23 @@
+FROM continuumio/miniconda:latest
```
[the remark this answers is not in the document]
Comment (b97pla) [reply docker/Dockerfile:1]: do you mean building and deploying the docker image to docker hub pulling from there when building the singularity image? I'm not sure having to maintain that outweighs the hassle of creating the docker image first (given that the use of this image would be very limited).. Or do you mean basically translating the Dockerfile to a Singularity recipe? I guess that could be done, it's just that my experience with singularity recipes is very limited so I stuck with the Dockerfile..

File: ngi_pipeline/tests/engines/sarek/models/test_sarek.py (hunk 1 above)
Comment (b97pla) [reply ngi_pipeline/tests/engines/sarek/models/test_sarek.py:127]: good point, I'll look into that

File: docker/ngi_config.yaml (hunk 3 above)
Comment (b97pla) [reply docker/ngi_config.yaml:33]: thanks, I'll update this config and also the upcoming PR to irma-provision when this has been merged

File: requirements-dev.txt (hunk 2 above)
Comment (b97pla) [reply requirements-dev.txt:2]: ok, this was probably a mistake. PyCharm was complaining a lot about old dependencies but I think I also had some problem with incompatible dependencies when creating the docker image and taking away the versions fixed that. I'll go over it again and see if I can fix it.

**Comment by b97pla** [comment]
Thanks for the reviews and apologies for the size of the PR. 

I have restored the versions on the dependencies and fixed the conflicts, as well as refactored the test. I also fixed some of the code complexity that CodeClimate was complaining about.

File: docker/Dockerfile (hunk 4 above)
[the remark this answers is not in the document]
Comment (matrulda) [reply docker/Dockerfile:1]: I meant translating it to a singularity recipe. The docker image takes up some space. But it's not that important.

**Review by matrulda — APPROVED** [review]
Nice, looks good to me!
